### PR TITLE
Support hessian in host sketch container.

### PR DESCRIPTION
**Pull request by trivialfis**
* Implement some utilities for threads to reduce dependency on `omp_get_max_threads`.
* Merge hessian with sample weights.

### CODE REVIEW

File: tests/cpp/common/test_hist_util.cc [hunk 1]
```diff
@@ -226,6 +226,26 @@ TEST(HistUtil, DenseCutsAccuracyTestWeights) {
   }
 }
 
+TEST(HistUtil, QuantileWithHessian) {
```
Comment (trivialfis) [inline tests/cpp/common/test_hist_util.cc:229]: Compare the result with multiplying Hessian directly.

**Comment by trivialfis** [comment]
This is part of refactoring the approx tree method.  In the implementation of current `approx` the sample weight is ignored, but the impl in this PR multiples the sample weight with hessian.

File: src/common/quantile.cc [hunk 2]
```diff
@@ -89,40 +91,94 @@ std::vector<bst_feature_t> HostSketchContainer::LoadBalance(
...
+    auto const &group_weights = info.weights_.HostVector();
+    CHECK_GE(group_ptr.size(), 2);
+    CHECK_EQ(group_ptr.back(), hessian.size());
+    size_t cur_group = 0;
+    for (size_t i = 0; i < hessian.size(); ++i) {
+      results[i] = hessian[i] * group_weights[cur_group];
+      if (i == group_ptr[cur_group + 1]) {
+        cur_group++;
+      }
+    }
```
Comment (hcho3) [inline src/common/quantile.cc:112]: This portion is sequential for now?

**Review by hcho3 — APPROVED** [review]

File: src/common/quantile.cc (hunk 2 above)
Comment (trivialfis) [reply src/common/quantile.cc:112]: It's either sequential or binary search.

**Pull request merged**